core/monitoring/power_monitor.py

```
@@ -0,0 +1,417 @@
+                return self._capture_nut_power()
+            elif ups_type == "cyberpower":
+                return self._capture_cyberpower_power()
+        except Exception as e:
+            print(f"Error reading UPS: {e}")
+        
+        return None
+    
+    def _capture_apc_power(self) -> Optional[Dict]:
+        """Capture power from APC UPS via apcaccess."""
+        try:
+            result = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call with `shell=True` seems safe, but may be changed in the future; consider rewriting without `shell`

(S602)